### PR TITLE
refactor(models): adjust testing mocks to valid only

### LINKED ISSUE #173: Adjust testing strategy for fixture values
The fixtures should reflect the parsing logic and edge cases should be implemented visibly as over write.

Before:

```typescript

export function config(outputDir = 'tmp'): CoreConfig {
  return {
    persist: { outputDir },
    // ...
  }
};


 it('should throw if outputDir is invalid', () => {
    const persistConfigMock = persistConfig({outputDir : ' '});

    expect(() => persistConfigSchema.parse(persistConfigMock)).toThrow(
      `path is invalid`,
    );
  });
```

This is not in sync with the logic like default, validation etc in schemas. A change later already fail in the mock and we have to use visible overwrites for invalid data.

After:

```typescript

export function config(outputDir = 'tmp'): CoreConfig {
  return {
    persist: persistConfigSchema.parse({ outputDir }),
    // ...
  }
};


 it('should throw if outputDir is invalid', () => {
    const persistConfigMock = persistConfig();
    persistConfigMock.outputDir = ' ';

    expect(() => persistConfigSchema.parse(persistConfigMock)).toThrow(
      `path is invalid`,
    );
  });
```

### PR DESCRIPTION
**Pull request by BioPhoton**
This PR includes a couple of file splittings and new types to get rid of the intermediate type hacks in models.

It also closes #173

### CODE REVIEW

File: packages/models/src/lib/implementation/schemas.ts [hunk 1]
```diff
@@ -150,6 +151,7 @@ export function weightedRefSchema(
     { description },
   );
 }
+export type WeightedRef = z.infer<ReturnType<typeof weightedRefSchema>>; // @TODO whi is slug optional? @matej
```
[the remark this answers is not in the document]
Comment (BioPhoton) [reply packages/models/src/lib/implementation/schemas.ts:168]: i will figure it out...

File: tmp/.gitkeep
[the remark this answers is not in the document]
Comment (BioPhoton) [reply tmp/.gitkeep:1]: will get addressed after the filename PR of even in this PR

File: packages/models/src/lib/implementation/schemas.ts [hunk 2]
```diff
@@ -168,6 +169,7 @@ export function weightedRefSchema(
...
```
Comment (BioPhoton) [inline packages/models/src/lib/implementation/schemas.ts:172]:
```suggestion
export type WeightedRef = z.infer<ReturnType<typeof weightedRefSchema>>; // @TODO why is slug optional? @matej
```

File: packages/models/src/lib/implementation/schemas.ts [hunk 3]
```diff
@@ -168,6 +169,7 @@ export function weightedRefSchema(
     { description },
   );
 }
+export type WeightedRef = z.infer<ReturnType<typeof weightedRefSchema>>; // @TODO why is slug optional? @matej
```
Comment (matejchalk) [inline packages/models/src/lib/implementation/schemas.ts:172]: Just checked the code - it isn't.

```suggestion
export type WeightedRef = z.infer<ReturnType<typeof weightedRefSchema>>;
```

**Review by matejchalk — APPROVED** [review]

**Pull request merged**